src/extensions/nodejsMigrationHelper.ts

```diff
@@ -0,0 +1,75 @@
+import { FirebaseError } from "../error";
+import * as extensionsApi from "./extensionsApi";
+import { logPrefix } from "./extensionsHelper";
+import { promptOnce } from "../prompt";
+import * as utils from "../utils";
+
+const changeMsg =
+  "Node.js 8 has been deprecated. It’s recommended to update this extension to ensure it is running on Node.js 10.";
+const billingMsgUpdate =
+  "This update includes an upgrade to Node.js 10 from Node.js 8, which is no longer maintained. Your project must be on the Blaze (pay as you go) plan to deploy Node.js 10 functions. Starting with this update, you will be charged a small amount when you deploy this extension, including when you make configuration changes and apply future updates.";
+const billingMsgCreate =
+  "This extension includes Node.js 10 functions. Your project must be on the Blaze (pay as you go) plan to deploy Node.js 10 functions. You will be charged a small amount when you deploy this extension, including when you make configuration changes and apply future updates.";
+
+function hasRuntime(resource: extensionsApi.Resource, runtime: string): boolean {
+  let resourceRuntime = "nodejs8"; // default to nodejs8 runtime
+  if (resource.properties && resource.properties.runtime) {
+    resourceRuntime = resource.properties.runtime;
+  }
+  return resourceRuntime == runtime;
+}
+
+function shouldDisplayMsg(
+  newSpec: extensionsApi.ExtensionSpec,
+  curSpec?: extensionsApi.ExtensionSpec
+): boolean {
+  const newResources = newSpec.resources;
+  const curResources = curSpec?.resources || [];
+  return (
+    newResources.some((r) => hasRuntime(r, "nodejs10")) &&
```

Comment:
Optional: There is an edge case here, where if someone made an extension with a mix of node8 and node 10 functions, it would always show this message on updates, even if no runtimes changed. I don't think this is a meaningful edge case or requires change, but if you can think of a nice way to handle it, we may as well.

Reply:
Ack. I don't think we'll ever see the edge case you described while we have control over all published extensions.